Does not build on Node 12

<!--

Have you read Atom's Code of Conduct? By filing an Issue, you are expected to comply with it, including treating everyone with respect: https://github.com/atom/.github/blob/master/CODE_OF_CONDUCT.md

Do you want to ask a question? Are you looking for support? The Atom message board is the best place for getting support: https://discuss.atom.io

-->

### Prerequisites

* [x] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: <https://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode>
    * Followed all applicable steps in the debugging guide: <https://flight-manual.atom.io/hacking-atom/sections/debugging/>
    * Checked the FAQs on the message board for common solutions: <https://discuss.atom.io/c/faq>
    * Checked that your issue isn't already filed: <https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom>
    * Checked that there is not already an Atom package that provides the described functionality: <https://atom.io/packages>

### Description

<!-- Description of the issue -->

### Steps to Reproduce

1. Install Node.js 12 
2. Clone this repository (`git clone https://github.com/atom/spawn-as-admin.git`)
3. Navigate to it via command line (`cd spawn-as-admin`)
4. Run `npm install` to build

**Expected behavior:**

Dependencies install and the project builds without issue

**Actual behavior:**

Fails with C++ compilation errors. Full logs attached, partial log follows

```
> spawn-as-admin@0.1.3 install /Users/palette/Development/spawn-as-admin
> node-gyp rebuild

  CXX(target) Release/obj.target/spawn_as_admin/src/main.o
In file included from ../src/main.cc:1:
In file included from ../node_modules/nan/nan.h:192:
../node_modules/nan/nan_maybe_43_inl.h:112:15: error: no member named 'ForceSet' in 'v8::Object'
  return obj->ForceSet(isolate->GetCurrentContext(), key, value, attribs);
         ~~~  ^
In file included from ../src/main.cc:1:
In file included from ../node_modules/nan/nan.h:197:
In file included from ../node_modules/nan/nan_converters.h:59:
../node_modules/nan/nan_converters_43_inl.h:22:1: warning: 'ToBoolean' is deprecated: ToBoolean can never throw. Use Local version. [-Wdeprecated-declarations]
X(Boolean)
^
../node_modules/nan/nan_converters_43_inl.h:18:12: note: expanded from macro 'X'
      val->To ## TYPE(v8::Isolate::GetCurrent()->GetCurrentContext())          \
           ^

< ... snipped ... >

17 warnings and 16 errors generated.
make: *** [Release/obj.target/spawn_as_admin/src/main.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/palette/.nvs/node/12.13.0/x64/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:210:5)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)
gyp ERR! System Darwin 19.0.0
gyp ERR! command "/Users/palette/.nvs/node/12.13.0/x64/bin/node" "/Users/palette/.nvs/node/12.13.0/x64/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/palette/Development/spawn-as-admin
gyp ERR! node -v v12.13.0
gyp ERR! node-gyp -v v5.0.5
gyp ERR! not ok 
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! spawn-as-admin@0.1.3 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the spawn-as-admin@0.1.3 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/palette/.npm/_logs/2019-10-28T14_30_20_237Z-debug.log
```

**Reproduces how often:**

100%

### Versions

* macOS Catalina 10.15 (19A602)
* Node.js 12.13.0

### Additional Information

<!-- Any additional information, configuration or data that might be necessary to reproduce the issue. -->

[npm_install.log](https://github.com/atom/spawn-as-admin/files/3779151/npm_install.log)

same questions

```
E:\workspaces\webstorm\electron-demo\node_modules\spawn-as-admin>node-gyp rebuild --arch=ia32 --target=12.8.1
gyp info it worked if it ends with ok
gyp info using node-gyp@6.0.0
gyp info using node@10.15.3 | win32 | x64
gyp info find Python using Python version 2.7.15 found at "C:\Python27\python.exe"
gyp info find VS using VS2017 (15.9.28307.858) found at:
gyp info find VS "C:\Program Files (x86)\Microsoft Visual Studio\2017\Community"
gyp info find VS run with --verbose for detailed information
gyp info spawn C:\Python27\python.exe
gyp info spawn args [ 'C:\\Users\\lishichao\\AppData\\Roaming\\npm\\node_modules\\node-gyp\\gyp\\gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'msvs',
gyp info spawn args   '-I',
gyp info spawn args   'E:\\workspaces\\webstorm\\electron-demo\\node_modules\\spawn-as-admin\\build\\config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Users\\lishichao\\AppData\\Roaming\\npm\\node_modules\\node-gyp\\addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Users\\lishichao\\AppData\\Local\\node-gyp\\Cache\\12.8.1\\include\\node\\common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=C:\\Users\\lishichao\\AppData\\Local\\node-gyp\\Cache\\12.8.1',
gyp info spawn args   '-Dnode_gyp_dir=C:\\Users\\lishichao\\AppData\\Roaming\\npm\\node_modules\\node-gyp',
gyp info spawn args   '-Dnode_lib_file=C:\\\\Users\\\\lishichao\\\\AppData\\\\Local\\\\node-gyp\\\\Cache\\\\12.8.1\\\\<(target_arch)\\\\node.lib',
gyp info spawn args   '-Dmodule_root_dir=E:\\workspaces\\webstorm\\electron-demo\\node_modules\\spawn-as-admin',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'E:\\workspaces\\webstorm\\electron-demo\\node_modules\\spawn-as-admin\\build',
gyp info spawn args   '-Goutput_dir=.' ]
?[37;40mgyp?[0m ?[0m?[32minfo?[0m ?[0m?[35mspawn?[0m C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin\MSBuild.exe
?[0m?[37;40mgyp?[0m ?[0m?[32minfo?[0m ?[0m?[35mspawn args?[0m [ 'build/binding.sln',
?[0m?[37;40mgyp?[0m ?[0m?[32minfo?[0m ?[0m?[35mspawn args?[0m   '/clp:Verbosity=minimal',
?[0m?[37;40mgyp?[0m ?[0m?[32minfo?[0m ?[0m?[35mspawn args?[0m   '/nologo',
?[0m?[37;40mgyp?[0m ?[0m?[32minfo?[0m ?[0m?[35mspawn args?[0m   '/p:Configuration=Release;Platform=Win32' ]
?[0m在此解决方案中一次生成一个项目。若要启用并行生成，请添加“/m”开关。
  main.cc
  spawn_as_admin_win.cc
  win_delay_load_hook.cc
e:\workspaces\webstorm\electron-demo\node_modules\spawn-as-admin\src\main.cc(26): warning C4996: 'Nan::Callback::Call': 被声明为已否决 [E:\workspaces\webstorm\electron-demo\node_modules\spawn-as-admin\build\spawn_as_admin.vcxproj]
  e:\workspaces\webstorm\electron-demo\node_modules\nan\nan.h(1740): note: 参见“Nan::Callback::Call”的声明
e:\workspaces\webstorm\electron-demo\node_modules\spawn-as-admin\src\main.cc(36): error C2440: “<function-style-cast>”: 无法从“v8::Local<v8::Value>”转换为“v8::String::Utf8Value” [E:\workspaces\webstorm\electron-demo\node_modules\spawn-as-admin\build\spawn_as_admin.vcxproj]
  e:\workspaces\webstorm\electron-demo\node_modules\spawn-as-admin\src\main.cc(36): note: 无构造函数可以接受源类型，或构造函数重载决策不明确
e:\workspaces\webstorm\electron-demo\node_modules\spawn-as-admin\src\main.cc(53): error C2440: “<function-style-cast>”: 无法从“v8::Local<v8::Value>”转换为“v8::String::Utf8Value” [E:\workspaces\webstorm\electron-demo\node_modules\spawn-as-admin\build\spawn_as_admin.vcxproj]
  e:\workspaces\webstorm\electron-demo\node_modules\spawn-as-admin\src\main.cc(53): note: 无构造函数可以接受源类型，或构造函数重载决策不明确
e:\workspaces\webstorm\electron-demo\node_modules\spawn-as-admin\src\main.cc(47): warning C4996: 'v8::Object::Get': 被声明为已否决 [E:\workspaces\webstorm\electron-demo\node_modules\spawn-as-admin\build\spawn_as_admin.vcxproj]
  c:\users\lishichao\appdata\local\node-gyp\cache\12.8.1\include\node\v8.h(3423): note: 参见“v8::Object::Get”的声明
e:\workspaces\webstorm\electron-demo\node_modules\spawn-as-admin\src\main.cc(68): warning C4996: 'v8::Object::Set': 被声明为已否决 [E:\workspaces\webstorm\electron-demo\node_modules\spawn-as-admin\build\spawn_as_admin.vcxproj]
  c:\users\lishichao\appdata\local\node-gyp\cache\12.8.1\include\node\v8.h(3365): note: 参见“v8::Object::Set”的声明
e:\workspaces\webstorm\electron-demo\node_modules\spawn-as-admin\src\main.cc(69): warning C4996: 'v8::Object::Set': 被声明为已否决 [E:\workspaces\webstorm\electron-demo\node_modules\spawn-as-admin\build\spawn_as_admin.vcxproj]
  c:\users\lishichao\appdata\local\node-gyp\cache\12.8.1\include\node\v8.h(3365): note: 参见“v8::Object::Set”的声明
e:\workspaces\webstorm\electron-demo\node_modules\spawn-as-admin\src\main.cc(70): warning C4996: 'v8::Object::Set': 被声明为已否决 [E:\workspaces\webstorm\electron-demo\node_modules\spawn-as-admin\build\spawn_as_admin.vcxproj]
  c:\users\lishichao\appdata\local\node-gyp\cache\12.8.1\include\node\v8.h(3365): note: 参见“v8::Object::Set”的声明
e:\workspaces\webstorm\electron-demo\node_modules\spawn-as-admin\src\main.cc(76): error C2065: “Handle”: 未声明的标识符 [E:\workspaces\webstorm\electron-demo\node_modules\spawn-as-admin\build\spawn_as_admin.vcxproj]
e:\workspaces\webstorm\electron-demo\node_modules\spawn-as-admin\src\main.cc(76): error C2275: “v8::Object”: 将此类型用作表达式非法 [E:\workspaces\webstorm\electron-demo\node_modules\spawn-as-admin\build\spawn_as_admin.vcxproj]
  c:\users\lishichao\appdata\local\node-gyp\cache\12.8.1\include\node\v8.h(3363): note: 参见“v8::Object”的声明
e:\workspaces\webstorm\electron-demo\node_modules\spawn-as-admin\src\main.cc(76): error C2065: “exports”: 未声明的标识符 [E:\workspaces\webstorm\electron-demo\node_modules\spawn-as-admin\build\spawn_as_admin.vcxproj]
e:\workspaces\webstorm\electron-demo\node_modules\spawn-as-admin\src\main.cc(76): error C2182: “Init”: 非法使用“void”类型 [E:\workspaces\webstorm\electron-demo\node_modules\spawn-as-admin\build\spawn_as_admin.vcxproj]
e:\workspaces\webstorm\electron-demo\node_modules\spawn-as-admin\src\main.cc(76): error C2448: “spawn_as_admin::Init”: 函数样式初始值设定项类似函数定义 [E:\workspaces\webstorm\electron-demo\node_modules\spawn-as-admin\build\spawn_as_admin.vcxproj]
?[37;40mgyp?[0m ?[0m?[31;40mERR!?[0m ?[0m?[35mbuild error?[0m
?[0m?[37;40mgyp?[0m ?[0m?[31;40mERR!?[0m ?[0m?[35mstack?[0m Error: `C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin\MSBuild.exe` failed with exit code: 1
?[0m?[37;40mgyp?[0m ?[0m?[31;40mERR!?[0m ?[0m?[35mstack?[0m     at ChildProcess.onExit (C:\Users\lishichao\AppData\Roaming\npm\node_modules\node-gyp\lib\build.js:194:23)
?[0m?[37;40mgyp?[0m ?[0m?[31;40mERR!?[0m ?[0m?[35mstack?[0m     at ChildProcess.emit (events.js:189:13)
?[0m?[37;40mgyp?[0m ?[0m?[31;40mERR!?[0m ?[0m?[35mstack?[0m     at Process.ChildProcess._handle.onexit (internal/child_process.js:248:12)
?[0m?[37;40mgyp?[0m ?[0m?[31;40mERR!?[0m ?[0m?[35mSystem?[0m Windows_NT 10.0.15063
?[0m?[37;40mgyp?[0m ?[0m?[31;40mERR!?[0m ?[0m?[35mcommand?[0m "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\lishichao\\AppData\\Roaming\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild" "--arch=ia32" "--target=12.8.1"
?[0m?[37;40mgyp?[0m ?[0m?[31;40mERR!?[0m ?[0m?[35mcwd?[0m E:\workspaces\webstorm\electron-demo\node_modules\spawn-as-admin
?[0m?[37;40mgyp?[0m ?[0m?[31;40mERR!?[0m ?[0m?[35mnode -v?[0m v10.15.3
?[0m?[37;40mgyp?[0m ?[0m?[31;40mERR!?[0m ?[0m?[35mnode-gyp -v?[0m v6.0.0
?[0m?[37;40mgyp?[0m ?[0m?[31;40mERR!?[0m ?[0m?[35mnot ok?[0m
?[0m
```

--- a/package.json
+++ b/package.json
@@ -18,7 +18,7 @@
     "test": "mocha --ui=tdd test/*.test.js"
   },
   "dependencies": {
-    "nan": "2.x"
+    "nan": "^2.14.0"
   },
   "devDependencies": {
     "mocha": "^3.5.0"

--- a/src/main.cc
+++ b/src/main.cc
@@ -33,7 +33,7 @@ void SpawnAsAdmin(const Nan::FunctionCallbackInfo<Value>& info) {
     return;
   }
 
-  std::string command(*String::Utf8Value(info[0]));
+  std::string command(*Nan::Utf8String(info[0]));
 
   if (!info[1]->IsArray()) {
     Nan::ThrowTypeError("Arguments must be an array");
@@ -50,7 +50,7 @@ void SpawnAsAdmin(const Nan::FunctionCallbackInfo<Value>& info) {
       return;
     }
 
-    args.push_back(std::string(*String::Utf8Value(js_arg)));
+    args.push_back(std::string(*Nan::Utf8String(js_arg)));
   }
 
   if (!info[2]->IsFunction()) {
@@ -73,7 +73,7 @@ void SpawnAsAdmin(const Nan::FunctionCallbackInfo<Value>& info) {
   Nan::AsyncQueueWorker(new Worker(new Nan::Callback(info[2].As<Function>()), child_process, test_mode));
 }
 
-void Init(Handle<Object> exports) {
+void Init(Local<Object> exports) {
   Nan::SetMethod(exports, "spawnAsAdmin", SpawnAsAdmin);
 }
 

